Automattic/wp-calypso
# Register all store routes needed by nav

Fixes #13620 

This PR adds registration of all the routes needed by the store dashboard and store nav (see #13466 for design).  For routes without containers yet, the Dashboard component is used as a placeholder.

To test:

Ensure the following URLs load the Dashboard
http://calypso.localhost:3000/store/{storeslug}
http://calypso.localhost:3000/store/products/{storeslug}
http://calypso.localhost:3000/store/products/{storeslug}/import
http://calypso.localhost:3000/store/orders/{storeslug}
http://calypso.localhost:3000/store/orders/{storeslug}/add
http://calypso.localhost:3000/store/promotions/{storeslug}
http://calypso.localhost:3000/store/extensions/{storeslug}
http://calypso.localhost:3000/store/settings/{storeslug}
http://calypso.localhost:3000/store/settings/{storeslug}/checkout
http://calypso.localhost:3000/store/settings/{storeslug}/shipping
http://calypso.localhost:3000/store/settings/{storeslug}/tax

Ensure the following URL loads ProductCreate
http://calypso.localhost:3000/store/products/{storeslug}/add

Note that Stats is also connected in the code, but config.isEnabled false
http://calypso.localhost:3000/store/stats/{storeslug}

The idea is that as developers add new containers, they need only add an import statement for it and then update the appropriate element in `storePages`

If reviewers would prefer, I could create a throwaway Placeholder component and have all the non-dashboard routes reference that instead. Just let me know.

cc @coderkevin @justinshreve @belcherj @jeffstieler @marcinbot @DanReyLop
